addons/test/cert-test.c

```diff
@@ -70,3 +70,45 @@ void str05()
     wchar_t *str2  = L"hello";  //cert-STR05-C
 }
 
+void msc24()
+{
+    time_t rawtime;
+    struct tm *timeinfo;
+    char buffer[256];
+    int i;
+    long int li;
+    long long int lli;
+    FILE *f;
+
+    f = fopen ("myfile.txt","w+");  //cert-MSC24-C
+    setbuf ( f , buffer )   //cert-MSC24-C
+    for ( i='A' ; i<='Z' ; i++)
+        fputc ( n, f);
+    rewind (f);             //cert-MSC24-C
+    fread (buffer,1,26,f);
+    fclose (f);
+
+    time ( &rawtime );
+    timeinfo = localtime ( &rawtime );
+    printf ( "The current date/time is: %s", asctime (timeinfo) ); //cert-MSC24-C
+
+    strcpy(buffer, "2.5");          
```

Comment:
You don't have to write additional tests ... but I would prefer that there was no warning for this code:

    struct S {
        int x; int strcpy;
    };
   struct S s;
   s.strcpy = 123;

or this code:

    strcpy(a,b,c);

Reply:
The MSC24-C check doesn't check for strcpy so I don't get any warnings when I add your example lines to the cert-test.c for this check.

Reply:
ok my mistake.. replace "strcpy" with an arbitrary function that is checked then.. atof for instance.